### PR TITLE
Remove "| O_BINARY" or " |= O_BINARY" since O_BINARY is 0 now

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
follow nuttx kernel side change: https://github.com/apache/incubator-nuttx/pull/6357

## Impact
Minor

## Testing
Pass CI

### CODE REVIEW

**Comment by davids5** [comment]
This change hides the intent the the file is opened in binary mode. Does it make sense to assume the value the flag has?

**Comment by xiaoxiang781216** [comment]
> This change hides the intent the the file is opened in binary mode. Does it make sense to assume the value the flag has?

Most caller of open want the binary mode behavior:
https://github.com/apache/incubator-nuttx-apps/search?q=open
but only these two places add O_BINARY. So the implementation of open need hold the following assumption:
File must be opened in binary mode if both O_BINARY and O_TEXT isn't passed to open.
Actually, other POSIX compliant OS doesn't support O_BINAY and O_TEXT at all:
https://stackoverflow.com/questions/2266992/no-o-binary-and-o-text-flags-in-linux
So, this change just make the code more consistent.

**Comment by pkarashchenko** [comment]
@davids5 are you ok with this PR and PR to the OS side?

**Comment by pkarashchenko** [comment]
In general there is no harm in keeping the binary flag in code even if it is 0. But we can remove it of course.

**Comment by davids5** [comment]
@pkarashchenko Thank you for asking. I see it as a breaking changes, that does not add much value. But at least the compile will fail and trigger the hunt for "What! Why is NuttX not building again?"

**Comment by xiaoxiang781216** [comment]
> @pkarashchenko Thank you for asking. I see it as a breaking changes, that does not add much value. But at least the compile will fail and trigger the hunt for "What! Why is NuttX not building again?"

It isn't a breaking change: both flag still exist. I modify these two files just want to make sure the assumption more clear and consistent with the rest code in the repo.

**Comment by davids5** [comment]
@xiaoxiang781216 Thank you for detailed explanation. It would be appreciated if your PR would include that sort of information when submitted.

File: netutils/ftpd/ftpd.c [hunk 1]
```diff
@@ -1809,9 +1809,6 @@ static int ftpd_stream(FAR struct ftpd_session_s *session, int cmdtype)
 #if defined(O_LARGEFILE)
   oflags |= O_LARGEFILE;
 #endif
-#if defined(O_BINARY)
```
Comment (pkarashchenko) [inline netutils/ftpd/ftpd.c:1812]: Let's check the original canutils code. If it contains this code, then it is better to keep it even if the binary flag is zero

Comment (xiaoxiang781216) [reply netutils/ftpd/ftpd.c:1812]: Sure

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 Thank you for detailed explanation. It would be appreciated if your PR would include that sort of information when submitted.

Ok.

File: netutils/ftpd/ftpd.c (hunk 1 above)
Comment (pkarashchenko) [reply netutils/ftpd/ftpd.c:1812]: Posted comment on a wrong line, but still... I've checked can-utils and do not see any reference to O_BINARY

**Review by pkarashchenko — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**